homeassistant/components/recorder/__init__.py

```diff
@@ -982,6 +1018,41 @@ def block_till_done(self):
         self.queue.put(WaitTask())
         self._queue_watch.wait()
 
+    async def lock_database(self) -> bool:
+        """Lock database so it can be backed up safely."""
+        if self._database_lock_task:
+            _LOGGER.warning("Database already locked.")
+            return False
+
+        task = DatabaseLockTask(threading.Event(), threading.Event(), False)
```

Comment:
We could make it wait for an `asyncio.Event` and await that instead of typing up an executor thread

Reply:
`asyncio.Event` is not thread safe, and afaict we are running in two different threads...

Reply:
Rough implementation (untested) that avoids holding executor thread:
https://github.com/bdraco/home-assistant/commit/3f8a46c6f595830c00f7d1664d6a84387b7b77b8

Reply:
Hm, I see that way around. Yeah seems safe, and avoids `run_in_executor`. I'll test it and add it to this PR.